### PR TITLE
Repo Gardening: log failures to send Slack messages

### LINKED ISSUE #40366: Repo Gardening / Slack messages: detect when a message cannot be delivered
We currently do not do anything special with our requests to post Slack messages:
https://github.com/Automattic/jetpack/blob/199f5062ed30855d6344746b669a831fc5289a19/projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js#L91

https://github.com/Automattic/jetpack/blob/befb138355b9fd5483d9d0fd1f1b41f9d565f2a9/projects/github-actions/repo-gardening/src/tasks/flag-oss/index.js#L47-L51

Since we're posting to more and more channels, it would be nice to detect when a request has failed (for example because a channel has been archived) and warn the team in charge of the Repo Gardening action so they can update the channels where messages get posted.

Slack's API surfaces that info quite nicely:
https://api.slack.com/methods/chat.postMessage
`is_archived`

********

Related conversation:
p1732723599754299-slack-C05KLDWRGAX

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #40366

## Proposed changes:

Until now, we used to try to send messages to Slack, and didn't do anything if the message could not be sent. We'll now log the message, either by sending a Slack message to myself, or the Quality team for folks outside of the Automattic organization.

This will allow us to better detect triage Slack actions that could not be completed, because a team no longer exists and their Slack channel has been archived for example.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* p1732723599754299-slack-C05KLDWRGAX

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

This can be tested in a fork. Here is an example:
https://github.com/jeherve/jetpack/issues/187

Here is the resulting message: p1732803634693709-slack-CN2FSK7L4

### CODE REVIEW

**Review by zinigor — APPROVED** [review]
I have no problems with this, except I don't know how to test it. But you have probably done it yourself.

File: projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js [hunk 1]
```diff
@@ -77,18 +79,71 @@ async function sendSlackMessage( message, channel, payload, customMessageFormat
...
-			Accept: 'application/json',
-		},
-	} );
+	try {
+		const slackRequest = await slackApi.chat.postMessage( slackMessage );
+		return !! slackRequest.ok;
+	} catch ( error ) {
+		// The request failed.
+		// At this point, we want to log specific types of errors (let's avoid noise by logging temporary errors for example).
+		if ( ! error.code === ErrorCode.PlatformError ) {
```
Comment (anomiex) [inline projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js:88]:
```suggestion
		if ( error.code !== ErrorCode.PlatformError ) {
```

File: projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js [hunk 2]
```diff
@@ -77,18 +79,71 @@ async function sendSlackMessage( message, channel, payload, customMessageFormat
...
+			return false;
+		}
+
+		// See the list of error messages here: https://api.slack.com/methods/chat.postMessage#errors
+		const errorMessage = error?.data?.error ?? 'Unknown error';
+
+		// Let's send a direct message to @jeherve about it, so we can investigate.
+		// For folks outside of Automattic, let's use the Quality team channel.
```
Comment (anomiex) [inline projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js:96]: Why not send all to the channel?

File: projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js (hunk 1 above)
Comment (jeherve) [reply projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js:88]: d'oh!
https://github.com/Automattic/jetpack/pull/40414

File: projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js (hunk 2 above)
Comment (jeherve) [reply projects/github-actions/repo-gardening/src/utils/slack/send-slack-message.js:96]: I think that's what we'll end up doing on the long term, but for now I'd like to monitor the assignment list closely, especially as more teams get re-orged.

**Pull request merged**